enumerable.rb

```diff
@@ -0,0 +1,116 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    var = to_a
+    result = []
+    i = 0
+    while i < var.length
+      result << yield(var[i])
+      i += 1
+    end
+    result
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    var = to_a
+    result = i = 0
+    while i < var.length
+      result = yield(var[i], i)
+      i += 1
+    end
+    result
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    my_each { |i| arr.push(i) if yield(i) }
+    arr
+  end
+
+  def my_all?
+    var = to_a
+    var.empty?
+    if block_given?
+      my_each { |i| return false unless yield(i) }
+      return true
+    end
+    var.include?(nil) || var.include?(false) ? false : true
+  end
+
+  def my_any?
+    var = to_a
+    if block_given?
+      var.my_each { |i| return true if yield(i) }
+    else
+      var.my_each { |i| return true if i.nil? == false and i != false }
+    end
+    false
+  end
+
+  def my_none?
+    var = to_a
+    if block_given?
+      var.my_each { |i| return false if yield(i) }
+    else
+      var.my_each { |i| return false if i.nil? == false and i != false }
+    end
+    true
+  end
```

Comment:
- [ ] should have working `my_none?`  identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_none?
> Make sure to test not only with arrays but also ranges

- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
> In this example we are using the Integer class but should work for any other class

```ruby
[1, 2.5, 'a', 9].my_none?(Integer) == [1, 2.5, 'a', 9].none?(Integer) => false
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/94221739-f36a1d00-fea8-11ea-8ec8-3f613066d5ce.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690539-130ba180-6f69-11ea-8bd0-6ca061ab05d2.png)

- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_none?(/z/) == %w[dog door rod blade].none?(/z/) => true
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/94221794-0d0b6480-fea9-11ea-8a22-c04947c38b2f.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690722-65e55900-6f69-11ea-8881-9afe27d263dc.png)

- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
[3,4,7,11].my_none?(3) == [3,4,7,11].none?(3) => false
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/94221838-24e2e880-fea9-11ea-9773-597ad3fc2bc5.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690915-a9d85e00-6f69-11ea-81b3-7adb07b281ca.png)

### [OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end
```

To make this work we have to use block.call(element) instead of yield(element) in my_any?